### PR TITLE
Add babel-plugin-rewire for function mocking

### PR DESCRIPTION
**Pull request by weikinhuang**

### CODE REVIEW

**Comment by nemtsov** [comment]
😍

**Comment by weikinhuang** [comment]
I will have to note that this will break the html code coverage reporter.

**Comment by weikinhuang** [comment]
https://github.com/speedskater/babel-plugin-rewire/issues/148

**Comment by weikinhuang** [comment]
https://github.com/istanbuljs/babel-plugin-istanbul/issues/22

File: index.js [hunk 1]
```diff
@@ -21,6 +21,7 @@ module.exports = function(context, opts) {
 
   if (env === 'test') {
     config.plugins.push(require('babel-plugin-istanbul').default);
```
Comment (hzoo) [inline index.js]: Should we separate these into 2 envs?

Comment (weikinhuang) [inline index.js]: well, we want to add both in the `test` env, I can add another condition within to only include `istanbul` when `COVERAGE=1`

Comment (hzoo) [inline index.js]: ok, it does make it slower and also can't debug

**Pull request merged**